### PR TITLE
Add failing test for projections used as union field type

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
From the experiment https://github.com/rust-lang/rust/pull/103985.

cc @compiler-errors 
r? @oli-obk

### CODE REVIEW

**Comment by oli-obk** [comment]
Was there an attempt to fix this by normalizing $somewhere?

File: tests/ui/projection-as-union-type.rs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+// This is currently not possible to use projections as union field's type.
```
[the remark this answers is not in the document]
Comment (fmease) [reply tests/ui/projection-as-union-type.rs:1]: Also it should either have an accompanying `.stderr` file or a `known-bug` header and no error annotations.

[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply tests/ui/projection-as-union-type.rs:1]: My rebase was a huge failure... ^^'

**Comment by GuillaumeGomez** [comment]
> Was there an attempt to fix this by normalizing $somewhere?

That's what I asked on zulip. I guess it should be normalized then. Investigating.

**Pull request closed without merging**